### DISCUSSION TITLE
Make the `getSingleResult()` add limit to the query

### DISCUSSION THREAD

**Discussion by francescortiz** [mikro-orm/mikro-orm#5379] [Ideas]
**Is your feature request related to a problem? Please describe.**
When you invoke `getSingleResult()` you expect `.limit(1)` to be added to the query. Since this is not happening, this leads to unforeseen performance issues.

**Describe the solution you'd like**
1. `.getSingleResult()` should add `.limit(1)` to the query.
2. `.execute('get')` shoudl add `.limit(1)` to the query.

Maybe the best approach can be to make `.getSingleResult()` with `limit == 1` call `.execute('get')` instead of `.execute('all')` and put this optimization only there.

**Additional context**
knex does this when you invoke [`.first()`](https://github.com/knex/knex/blob/aedba5e49ccbe4e44bf1a57f51a63d8b1ec0e6e6/lib/query/querybuilder.js#L1144)

**Comment by B4nan** [comment]
Generally speaking, there is a problem with `limit: 1` and loading to-many relations, there can't be any limit clause when this happens, as a single result can contain multiple rows on the raw result level. But I think we already detect this internally and remove the limit clause in such case.

> knex does this when you invoke [.first()](https://github.com/knex/knex/blob/aedba5e49ccbe4e44bf1a57f51a63d8b1ec0e6e6/lib/query/querybuilder.js#L1144)

Yes, in other words, you need to be explicit with knex, just like with MikroORM.

But I agree we should probably do this `qb.getSingleResult()` when there are no to-many joins in place. Will check if it won't break any tests.

**Reply by B4nan** [reply]
Yeah, looks like adding `this.limit(1)` to `getSingleResult` works fine and is correctly handled in case of a to-many join. There is one gotcha, the QB instance might be already finalized, that can happen if you first try to read the query and only then execute it, and we can't do much here (we could clone the QB, but then you would read the query and execute something else, which sounds just wrong).

And it looks like we can do the same for `qb.execute('get')`.

**Reply by B4nan** [reply]
Implemented via c2b22e80c47eabd2284dca41cedd7c871a78eac4
